src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.netstandard.cs

```diff
@@ -36,6 +39,30 @@ internal AssemblyName GetAssemblyName(StringHandle nameHandle, Version version,
             return assemblyName;
         }
 
+        internal static AssemblyName GetAssemblyName(string path)
```

Comment:
Private or internal problems tend to cause problems with linker, versioning, etc. Can this be a public method instead?

Reply:
It would be a new public API, isn't it? 

For now I just followed the pattern used in stack trace handling, but I did think about making the entrypoint meaningful  - so that we could make it public, if we'd like. I think we may even want to eventually deprecate the one on the `AssemblyName` and point users to this one.

Reply:
It would be good to make this public. If it needs to wait for API approval, it can be internal until approved. Either way.

Reply:
I agree that it would be nice to get it approved as a public API. 

Note that System.Reflection.Metadata is one of those assemblies that can be over-ridden by the app, so an app may be using a newer version of the assembly than what ships in the runtime.

Reply:
Right. The entrypoint is an internal compat contract even if this is not public. Another reason for making it public.

Reply:
Other thoughts I have - can `System.Reflection.Metadata` be older than needed or can it be missing?
Basically - do I need some special/graceful handling of situations when I can't bind to the helper?

The things like `throw new ArgumentException("cant get System.Reflection.Metadata.MetadataReader");` are placeholders. 
I think we can do a bit better, but not by much. 
The call will fail, but we can make diagnostics a bit friendlier.

Reply:
> can System.Reflection.Metadata be older than needed or can it be missing?

That would be user error - incorrectly constructed app.

> The things like throw new ArgumentException("cant get System.Reflection.Metadata.MetadataReader"); are placeholders.

I do think you need to add special messages. The default exception that you get for missing type should be enough.

Reply:
Ok, I'll let the error from `GetType` propagate and fro missing method I will throw `MissingMethodException`

Reply:
Entered an API proposal for the public helper - https://github.com/dotnet/runtime/issues/63960